### PR TITLE
feat: Presentation Exchange - BBS+ support

### PR DESCRIPTION
**Pull request by soluchok**
Adds BBS+ support for PE.

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/doc/presexch/definition.go [hunk 1]
```diff
@@ -298,8 +298,14 @@ func makeRequirement(requirements []*SubmissionRequirement, descriptors []*Input
 	return req, nil
 }
 
+// CredentialEntity contains the necessary information for processing the credential.
+type CredentialEntity struct {
```
Comment (llorllale) [inline pkg/doc/presexch/definition.go:302]: Can we rename this to "CredentialCandidate" to align a bit more with the text in the P-E spec? The term "candidate input" is used in section 5.3 but I think "CredentialCandidate" might be a better fit in this context.

Comment (soluchok) [reply pkg/doc/presexch/definition.go:302]: Sure.

File: pkg/doc/presexch/definition.go [hunk 2]
```diff
@@ -298,8 +298,14 @@ func makeRequirement(requirements []*SubmissionRequirement, descriptors []*Input
 	return req, nil
 }
 
+// CredentialCandidate contains the necessary information for processing the credential.
+type CredentialCandidate struct {
+	Opts       []verifiable.CredentialOpt
```
Comment (llorllale) [inline pkg/doc/presexch/definition.go:303]: I don't think we need to specify options for every credential. Can you explain why we do?

All `verifiable.CredentialOpts` should apply equally to all credentials. For example the JSONLD document loader configured in the unit test.

Comment (llorllale) [reply pkg/doc/presexch/definition.go:303]: I would probably refactor as: `CreateVP(credentials []*verifiable.Credential, options ...verifiable.CredentialOpt)`

Comment (soluchok) [reply pkg/doc/presexch/definition.go:303]: ok, I will rollback it. I did it bc I don't know what else might be needed :) working with structure is easier comparing to function arguments

Comment (llorllale) [reply pkg/doc/presexch/definition.go:303]: > comparing to function arguments

Ehm - to be clear, I was showing an example of how I would refactor the existing `CreateVP` function in my comment above. I was not suggesting functional arguments for `CreateVP`

Comment (soluchok) [reply pkg/doc/presexch/definition.go]: Done!

**Review by fqutishat — APPROVED** [review]

**Review by llorllale — APPROVED** [review]

**Pull request merged**